qdrant-landing/content/articles/qdrant-1.2.x.md

```diff
@@ -0,0 +1,206 @@
+
+```json
+{
+  "filter": {
+    "must": [
+      {
+        "key": "country.cities[].population",
+        "range": {
+          "gte": 2
+        }
+      },
+      {
+        "key": "country.cities[].area",
+        "range": {
+          "gt": 500,
+          "lte": 1000
+        }
+      }
+    ]
+  },
+  "limit": 3
+}
+```
+
+Japan would be returned because Tokyo and Osaka match the first criteria, while Kyoto fulfills 
+the second. But that's not what we wanted to achieve. That's the motivation behind introducing 
+a new type of nested filter.
+
+```json
+{
+  "filter": {
+    "must": [
+      {
+        "nested": {
+          "key": "country.cities",
+          "filter": {
+            "must": [
+              {
+                "key": "population",
+                "range": {
+                  "gte": 2
+                }
+              },
+              {
+                "key": "area",
+                "range": {
+                  "gt": 500,
+                  "lte": 1000
+                }
+              }
+            ]
+          }
+        }
+      }
+    ]
+  },
+  "limit": 3
+}
+```
+
+The syntax is consistent with all the other supported filters and enables new possibilities. In 
+our case, it allows us to express the joined condition on a nested structure and make the results 
+list empty but correct.
+
+## Important changes
+
+The latest release focuses not only on the new features but also introduces some changes making 
+Qdrant even more reliable.
+
+### Fixed memory errors on startup
+
+There has been an issue in memory-constrained environments, such as cloud, happening when users were 
+pushing massive amounts of data into the service using `wait=false`. This data influx resulted in an 
+overreaching of disk or RAM limits before the Write-Ahead Logging (WAL) was fully applied. This 
+situation was causing Qdrant to attempt a restart and reapplication of WAL, failing recurrently due 
+to the same memory constraints and pushing the service into a frustrating crash loop with many 
+Out-of-Memory errors.
+
+Qdrant 1.2 automatically detects if memory usage exceeds a specific threshold, informs a user with an 
+error message in the logs, and makes the service halt the loading of collection data and commence 
+operations in a partial state. This state allows for removing collections but doesn't support search 
+or update functions. If Qdrant enters this locked state, a reboot that either increases available or 
+reduces required resources resolves it.
+
+### Appendable mmap
+
+For a long time, segments using mmap storage were `non-appendable` and could only be constructed by 
+the optimizer. Dynamically adding vectors to the mmap file is fairly complicated and thus not 
+implemented in Qdrant, but we did our best to implement it in the recent release. If you want 
+to read more about segments, check out our docs on [vector storage](/documentation/storage/#vector-storage).
```

Comment:
Appendable mmap is indeed new in code, but if I am correct, there is no way for users to enable it at this moment.

Reply:
Users should be able to use it. This is a code problem. Looking into it now.

Reply:
I thought it was controlled by the optimizer. @generall Do you think we should keep that info?

Reply:
No I'm planning to work on that now.

I want to describe it in these two paragraphs:

- https://qdrant.tech/documentation/collections/#create-collection
- https://qdrant.tech/documentation/how_to/#bulk-upload-a-large-number-of-vectors

Reply:
Opened a PR for it here: https://github.com/qdrant/docs/pull/122